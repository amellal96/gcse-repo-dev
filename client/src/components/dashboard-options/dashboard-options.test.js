import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Link } from 'react-router-dom';

import DashboardOptions from './dashboard-options.component';

const mockStore = configureStore([thunk]);
jest.mock('react-router-dom/Link'); 

describe('Dashboard Options with student login', () => {
    let store;
    let wrapper;
    let mockHistory;
    let mockLink;

    beforeEach(() => {
        store = mockStore({
            user: {
                user: {
                    accountType: 'student'
                }
            }
        });

        store.dispatch = jest.fn();
        mockHistory = { listen: jest.fn() };
        mockLink = { to: jest.fn() };     

        wrapper = mount(
        <MemoryRouter >
                <Provider store={store}>
                    <DashboardOptions />
                </Provider>
        </MemoryRouter>
        );
    })

    it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render 2 dashboard options', () => {
        expect(wrapper.find('.dashboard-options')).toHaveLength(2);
    })
})