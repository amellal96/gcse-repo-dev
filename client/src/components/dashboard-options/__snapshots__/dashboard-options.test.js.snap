// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Options with student login should match the snapshot 1`] = `"<a href=\\"/\\"><div class=\\"options-container\\"><div class=\\"dashboard-option\\"><a href=\\"/browse\\"><div class=\\"option-text\\">Browse</div><img src=\\"square.png\\" alt=\\"\\"></a></div><div class=\\"dashboard-option\\"><a href=\\"/saved-questions\\"><div class=\\"option-text\\">Saved Questions</div><img src=\\"square.png\\" alt=\\"\\"></a></div></div></a>"`;
